plugins/c8y_log_plugin/src/main.rs

```diff
@@ -0,0 +1,240 @@
+            C8yTopic::SmartRestRequest.as_str(),
+        ));
+
+    let mqtt_client = mqtt_channel::Connection::new(&mqtt_config).await?;
+    Ok(mqtt_client)
+}
+
+pub async fn create_http_client(
+    tedge_config: &TEdgeConfig,
+) -> Result<JwtAuthHttpProxy, anyhow::Error> {
+    let mut http_proxy = JwtAuthHttpProxy::try_new(tedge_config).await?;
+    let () = http_proxy.init().await?;
+    Ok(http_proxy)
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let config_plugin_opt = LogfileRequestPluginOpt::parse();
+    let config_file = PathBuf::from(&format!(
+        "{:?}/{DEFAULT_TEDGE_CONFIG_PATH}",
+        &config_plugin_opt.config_dir
+    ));
+    let config_file_async_clone = config_file.clone();
+
+    let notify = Arc::new(Notify::new());
+    let notify_async_clone = notify.clone();
+
+    tedge_utils::logging::initialise_tracing_subscriber(config_plugin_opt.debug);
+
+    // Load tedge config from the provided location
+    let tedge_config_location =
+        tedge_config::TEdgeConfigLocation::from_custom_root(&config_plugin_opt.config_dir);
+    let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location.clone());
+    let tedge_config = config_repository.load()?;
+
+    let logs_dir = tedge_config.query(LogPathSetting)?;
+    let logs_dir = PathBuf::from(logs_dir.to_string());
+
+    if config_plugin_opt.init {
+        let () = init(&config_plugin_opt.config_dir, &logs_dir)?;
+        return Ok(());
+    }
+
+    // Create required clients
+    let mut mqtt_client = create_mqtt_client(&tedge_config).await?;
+    let mut http_client = create_http_client(&tedge_config).await?;
+
+    let plugin_config = Arc::new(PluginConfig::new(&config_file));
+    let msg = plugin_config.to_supported_config_types_message()?;
+    let () = mqtt_client.published.send(msg).await?;
+
+    tokio::task::spawn(async move {
+        let mut inotify = Inotify::init().expect("Error while initializing inotify instance");
+
+        // Watch for modify and close events.
+        inotify
+            .add_watch(
+                &config_file_async_clone,
+                WatchMask::MODIFY | WatchMask::CLOSE,
+            )
+            .expect("Failed to add file watch");
+
+        // Read events that were added with `add_watch` above.
+        let mut buffer = [0; 1024];
+
+        loop {
+            let events = inotify
+                .read_events_blocking(&mut buffer)
+                .expect("Error while reading events");
+
+            for event in events {
+                match event.mask {
+                    EventMask::CLOSE_WRITE => {
+                        notify_async_clone.notify_waiters();
+                    }
+                    _ => {}
+                }
+            }
+        }
+    });
+
+    loop {
+        tokio::select! {
+                message = mqtt_client.received.next() => {
+                let message = message.ok_or(anyhow!("Missing message."))?;
```

Comment:
Having no messages in not an error. The connection has been closed and one just has to break from the loop.